src/validation/permissions.js

```diff
@@ -0,0 +1,76 @@
+import Joi from '@hapi/joi';
+import { Role, Permission } from '../models';
+
+export const checkPermission = async (roleId, resourceId, permission) => {
+  try {
+    // Base case to check if there is no roleId given
+    if (!roleId) throw new Error('You are not authorized to perform this operation');
+    // Find Permission if exists
+    const foundPermission = await Permission.findOne({
+      attributes: [permission],
+      where: {
+        roleId,
+        resourceId
+      }
+    });
+    if (!foundPermission) {
+      throw new Error('You are not authorized to perform this operation');
+    }
+    // If Permission exists return the value of the permission
```

Comment:
Use js-doc instead

Reply:
The purpose of js-doc is for documentation. The above is just a comment to explain what is happening. I don't want to document if a variable exists or not